bluemira/equilibria/optimisation/problem/_tikhonov.py

```diff
@@ -183,10 +190,14 @@ def optimise(self, **_) -> CoilsetOptimiserResult:
         current_adjustment = tikhonov(a_mat, b_vec, self.gamma)
 
         # Update parameterisation (coilset).
-        current = self.coilset.get_control_coils().current + current_adjustment
-        self.coilset.get_control_coils().current = current
-        f_x = np.linalg.norm(a_mat @ current - b_vec) + np.linalg.norm(
-            self.gamma * current
+        opt_currents = self.coilset.get_control_coils().current + current_adjustment
+        self.coilset.set_optimisation_state(opt_currents=opt_currents, current_scale=1.0)
+
+        current = self.coilset.current
+
+        f_x = floatify(
+            np.linalg.norm(a_mat @ current - b_vec)
+            + np.linalg.norm(self.gamma * current)
```

Comment:
you could just use `.item()` here as you know it will be a numpy array. Up to you.